### PR TITLE
fix(ReadPreference): only allow valid ReadPreference modes

### PR DESCRIPTION
**Pull request by kvwalker**
Fixes NODE-1425

### CODE REVIEW

File: lib/core/topologies/read_preference.js [hunk 1]
```diff
@@ -12,6 +12,10 @@
  * @return {ReadPreference}
  */
 const ReadPreference = function(mode, tags, options) {
+  if (!ReadPreference.isValid(mode)) {
+    throw new TypeError('ReadPreference mode is not valid.');
```
Comment (mbroadst) [inline lib/core/topologies/read_preference.js]: Prefer a better error message:
```
throw new TypeError(`provided mode ${mode} is an invalid ReadPreference mode`);
```
or something along those lines.

File: lib/core/topologies/read_preference.js [hunk 2]
```diff
@@ -12,6 +12,10 @@
  * @return {ReadPreference}
  */
 const ReadPreference = function(mode, tags, options) {
```
Comment (mbroadst) [inline lib/core/topologies/read_preference.js]: While we're here, should we convert `ReadPreference` to a class?

File: test/functional/readpreference_tests.js [hunk 3]
```diff
@@ -546,4 +547,18 @@ describe('ReadPreference', function() {
...
     }
   });
+
+  it('Should throw an error on an invalid readPreference', function(done) {
+    const configuration = this.configuration;
+
+    const client = configuration.newClient();
+    client.connect((err, client) => {
+      const db = client.db(configuration.db);
+      expect(db.collection.bind(db, 'test', { readPreference: 'invalid' })).to.throw(
```
Comment (mbroadst) [inline test/functional/readpreference_tests.js]: did you mean to call `collection.bind` here?

File: lib/core/topologies/read_preference.js [hunk 4]
```diff
@@ -77,8 +81,8 @@ const VALID_MODES = [
   ReadPreference.SECONDARY,
   ReadPreference.SECONDARY_PREFERRED,
   ReadPreference.NEAREST,
-  true,
-  false,
+  // true,
```
Comment (daprahamian) [inline lib/core/topologies/read_preference.js]: should these be deleted

Comment (daprahamian) [reply lib/core/topologies/read_preference.js]: also, does removing these break any external workflows?

File: lib/core/topologies/read_preference.js (hunk 2 above)
Comment (kvwalker) [reply lib/core/topologies/read_preference.js]: I'd prefer to leave this work for another ticket and focus on Command Refactor work for now.

File: test/functional/readpreference_tests.js (hunk 3 above)
Comment (kvwalker) [reply test/functional/readpreference_tests.js]: Yes, so I can check that it is throwing the `readPreference` error.

File: lib/core/topologies/read_preference.js (hunk 4 above)
Comment (kvwalker) [reply lib/core/topologies/read_preference.js]: I don't think so and I think that `true` and `false` should not be considered valid `ReadPreference` modes anyways.

**Pull request merged**